src/webmock/core_ext.cr

```diff
@@ -1,14 +1,16 @@
 require "http/client"
 
 class HTTP::Request
-  getter scheme
-  setter scheme : String = "http"
+  property scheme : String = "http"
+
+  def full_uri
+    "#{scheme}://#{host_with_port}#{resource}"
+  end
 end
 
 class HTTP::Client
   private def exec_internal(request : HTTP::Request)
-    response = exec_internal(request) { |res| res }
-    response.tap do |response|
+    exec_internal(request, &.itself).tap do |response|
```

Comment:
Why there is a second argument in this invocation?

Reply:
To avoid going into the endless loop?

Reply:
@bcardiff ping